docs/testing.rst

```diff
@@ -83,23 +77,49 @@ In the above example the ``setUpClass`` is run twice, once for ``MyTestClass`` a
             ...
 
 However this can lead to giant Test classes. If you find that all the tests in a package or module are sharing
-the same set up, you can write a setup method for the entire package or module. More information on that can be found `here <http://pythontesting.net/framework/nose/nose-fixture-reference/#package>`_.
+the same set up, you can use a 'module' or 'package' `scoped fixture <https://github.com/dimagi/pytest-unmagic#fixture-scope>`_.
+
 
 Test tear down
-==================
+==============
 
-It is important to ensure that all objects you have created in the test database are deleted when the test
-class finishes running. This often happens in the ``tearDown`` method or the ``tearDownClass`` method.
-However, unneccessary cleanup "just to be safe" can add a large amount of time onto your tests.
+It is important to ensure that all objects you have created in databases are deleted when the test class finishes
+running. The best practice is to use pytest `fixtures <https://github.com/dimagi/pytest-unmagic>`_ or `addCleanup`
+or `addClassCleanup` to call cleanup routines. Cleanup can also be done in the ``tearDown`` method or the
+``tearDownClass`` method, although pytest fixtures and `add[Class]Cleanup` are preferred because, unlike
+`tearDown[Class]`, they get run if `setUp[Class]` raises an exception.
 
+SQL data in non-shard databases is automatically cleaned up on transaction rollback by test classes that override
+`django.test.TestCase` or function tests that use pytest-django's `db` fixture, so it is not necessary to delete
+SQL data in most cases. One exception to this is the shard dbs, since most usages of shard databases do not support
+transactions.
 
-Using SimpleTestCase
-====================
+Also beware that unneccessary cleanup "just to be safe" can add a large amount of time onto your tests and should
+be avoided.
+
+Functional database tests
+=========================
+
+Function tests may access databases by using pytest-django's ``db`` or ``transactional_db`` fixture. This is
+similar to extending ``django.test.TestCase`` or ``django.test.TransactionTestCase`` respectively.
+
+.. code:: python
+
+    from unmagic import use
+
+    @use("db")
```

Comment:
Just curious about the naming here.

How did you end up with the name "use" for this decorator?
"uses_fixtures" comes to mind as for something more descriptive.

Reply:
It is a succinct way to indicate that the test or fixture uses a fixture. To my knowledge it doesn't conflict with anything, and it's easy to tell what it does by looking at its docstring.

Reply:
Okay. 

Explicit naming just helps skipping the trip to docstring.